### DISCUSSION TITLE
questions related to groups discovery and new settings

### DISCUSSION THREAD

**Discussion by alka79** [fsaris/EspHome-AwoX-BLE-mesh-hub#72] [General]
Hello,
I had been using the previous version of the hub with great success. Very stable. Thanks for the good job :)
I clicked the upgrade button last week without paying much attention and faced instability of the awox devices. The debug log became very verbose with many reboots ( unhandled resets). I cleaned everything and reinstalled on a new board. Unfortunately no improvement. I tried again today and reinstalled once more: It now seems stable again with your latest changes.

Some questions :
context : I have 4 Awox bulbs and 1 Egloplug+. All in the same room. I also have 2 remotes. Each remote control the same group of the 4 bulbs.
In the Awox app, each of the 2 remotes defines a group named after the remote's name. There is no other group defined. 

- The hub reports 3 groups, named 0, 1, 3. Group0 seems to contain all my awox devices, and group 1 and 3 seem to match to the remotes. I would prefer to rely on HA to create groups. Would it be possible to make group discovery by the hub optional ? or give the option to filter out the groups we don't want discovered (maybe a whitelist for groups) ?

- There is a new sensor reported, with two entities. Binary sensor, category diagnostic. Named `connection 0 connected` and `connection 1 connected`. No manufacturer, no model. Quite obscure! One is almost always on while the second is off. What exactly do they mean ? Maybe a better naming ?

- I use the new mac address white list for my 5 devices, hoping it helps or speeds up discovery. (Previously the logs showed there were 7 devices found. Probably the two additional were the remotes but they were not recognized and seemed to create unnecessary burden.)
There is also a meshid white list. As I understand meshid and mac address are linked. What is the best practice to create a white list : meshids ? mac addresses? both ? 

- What exactly is the new setting `max-connections` ? (not knowing, I set it to 2)

Thanks again for the good job.
Al

**Comment by fsaris** [comment]
> * The hub reports 3 groups, named 0, 1, 3. Group0 seems to contain all my awox devices, and group 1 and 3 seem to match to the remotes. I would prefer to rely on HA to create groups. Would it be possible to make group discovery by the hub optional ? or give the option to filter out the groups we don't want discovered (maybe a whitelist for groups) ?

Was thinking about this, but not really sure what the benefits would be.
You could also easily disable the discovered groups in HA.

> * There is a new sensor reported, with two entities. Binary sensor, category diagnostic. Named `connection 0 connected` and `connection 1 connected`. No manufacturer, no model. Quite obscure! One is almost always on while the second is off. What exactly do they mean ? Maybe a better naming ?

This is related to `max-connections`.
This makes it visible if the hub has active connections. Mainly for diagnostics of the mesh

> * I use the new mac address white list for my 5 devices, hoping it helps or speeds up discovery. (Previously the logs showed there were 7 devices found. Probably the two additional were the remotes but they were not recognized and seemed to create unnecessary burden.)
>   There is also a meshid white list. As I understand meshid and mac address are linked. What is the best practice to create a white list : meshids ? mac addresses? both ?

The `meshids` limit the devices that are controlled by the hub. 
The `mac addresses` limit the devices that the hub tries to connect to. Related to the `max-connections` option.

With these options you can configure what devices the hub can connect and is suppose to control.
See #34 #12

> * What exactly is the new setting `max-connections` ? (not knowing, I set it to 2)

The max number of device connections the hub can create simultaneously. Mainly needed  when your devices/lights are not always online and/or the mesh network isn't fully reachable due to that #12

> Thanks again for the good job

Your welcome 👍🏻

**Comment by alka79** [comment]
Thanks for the detailed response. That clarifies :)
In my situation, all lamps in the same room and always powered, I will set both meshids and mac adresses white lists  and set max-connections to 1.

Abouts groups :
> You could also easily disable the discovered groups in HA

I did that. But at board reboot they come back ! It hurts my optimization nature as the groups topics create useless mqtt traffic. Anyway, it is not a big deal and I can leave with it.


On another optimization note : connection status topic is updated every 5 secs with the same payload, except timestamp. Maybe only publish an update when status changes ?
